cpp/src/barretenberg/stdlib/merkle_tree/nullifier_tree/nullifier_tree.hpp

```diff
@@ -25,12 +25,19 @@ template <typename Store> class NullifierTree : public MerkleTree<Store> {
 
     fr update_element(fr const& value);
 
-  private:
```

Comment:
Are we changing this from private to protected, only for testing purposes?

Reply:
Yes, because through the testing harness in aztec3 package we are exposing some of the internal values for testing purposes. But the original implementation of the testing harness used NullifierMemoryTree which was always only for testing so I can imagine that when it was originally implemented nobody cared about exposing more stuff.

Do you think we should rethink how we test it or is having some of the value as protected ok with you?

Reply:
Without looking deeper, it's okay with me. Just wanted to flag it up as usually changing the viability of an item for tests is a code smell.

Moreover, we discussed this being a problem in other parts of the code in Valencia, so this can be addressed when we look into cleaning up the API as a whole.

I'll open up an issue to track this when I get back to my laptop